File: Password Factory/Controls/Buttons/ZoomButton.m
//
//  ZoomButton.m
//  Password Factory
//
//  Created by Cristiana Yambo on 11/14/17.
//  Copyright © 2017 Cristiana Yambo. All rights reserved.
//

#import "ZoomButton.h"
#import "StyleKit.h"
#import "Utilities.h"
@implementation ZoomButton

-(instancetype)initWithCoder:(NSCoder *)coder {
    self = [super initWithCoder:coder];
    self.bordered = NO;
    self.imagePosition = NSImageOnly;
    [self setButtonType:NSMomentaryChangeButton];
    self.bezelStyle = NSRegularSquareBezelStyle;


    return self;
    
}
-(void)viewWillDraw {
    if ([Utilities isDarkMode]) {
        self.image = [StyleKit imageOfZoomWithZoomStroke:[NSColor whiteColor]];
        
    } else {
        self.image = [StyleKit imageOfZoomWithZoomStroke:[NSColor blackColor]];
    }
    [self setAlphaValue:0.5];
}
@end

File: Password Factory iOS/Controls/Buttons/ZoomButton.swift
//
//  ZoomButton.swift
//  Password Factory iOS
//
//  Created by Cristiana Yambo on 12/4/17.
//  Copyright © 2017 Cristiana Yambo. All rights reserved.
//

import UIKit
@IBDesignable

/// Zoom Button
class ZoomButton: UIButton {
    override func prepareForInterfaceBuilder() {
        super.prepareForInterfaceBuilder()
        setDisplay()
    }
    override func awakeFromNib() {
        super.awakeFromNib()
        setDisplay()
    }
    func setDisplay() {
        setImage(StyleKit.imageOfZoom(), for: .normal)
        setTitle("", for: .normal)
    }

}
